test/integration/stateView/applyStateTransitionFactory.spec.js

```diff
@@ -51,7 +54,7 @@ describe.skip('applyStateTransitionFactory', () => {
       null,
     );
 
-    dpp.setDataProvider(dataProvider);
+    dpp.dataProvider = dataProvider;
```

Comment:
why not just pass into the constructor as an option?

Reply:
`DataProvider` depends on `fetchContract`, which depends on `svContractMongoDbRepository`, which depends on `dpp`. Cyclic dependency.

Reply:
This won't work properly. We intentionally removed `setDataProvider`. You can pass a Data Provider only in the constructor. Let's just create one more DPP instance then in order to solve this cyclic problem?

Reply:
By the way, why do we need Data Provider here?